app/components/task-assignment.js

```diff
@@ -0,0 +1,87 @@
+import EmberCan from 'ember-can';
+import Ember from 'ember';
+import createTaskUserOptions from 'code-corps-ember/utils/create-task-user-options';
+
+const {
+  Component,
+  computed,
+  computed: { alias },
+  getProperties,
+  inject: { service },
+  set
+} = Ember;
+
+export default Component.extend({
+  classNames: ['task-assignment'],
+
+  currentUser: service(),
+  taskAssignment: service(),
+
+  task: null,
+  taskUser: null,
+  users: null,
+
+  // auto-assigns 'task' property from component as ability 'model'
+  ability: EmberCan.computed.ability('task'),
+  canAssign: alias('ability.canAssign'),
+  currentUserId: alias('currentUser.user.id'),
+  taskUserId: alias('taskUser.id'),
+
+  // TODO: this updates selection when it changes. However, it updates while
+  // the change is still processing, and rolls back if it fails.
+  // We should somehow skip the update if the change is processing
+  // TODO: It also fails to roll back when reassignment fails
+  didReceiveAttrs() {
```

Comment:
@daveconnis Could you make an issue for these two todos? Just describe where they are written in the code and copy them and the `didReceiveAttrs` function into the issue.

Generally, it's not a great approach to write todos in code, so if you see any in the future left by us, feel free to let us know by creating an issue.